Update deployer documentation (#390)
api3dao/airnode: docker/README.md

**This documentation focuses on the Deployer Docker image and its usage, not the Deployer itself. If you want learn more about the Deployer, please read [its documentation](../packages/deployer/README.md).**

# Build
You can build the Docker image by running:
```bash
docker build -f docker/Dockerfile -t api3/deployer:latest .
```

> Before the build, all Deployer dependencies (but not the Deployer itself) must be available in the package registry.

Author: I'd still keep it there even if just a note for us (or somebody new trying to build the container). But I'll add your suggestion to the section below.

# Configuration
## Credentials
In order to deploy Airnode to a cloud provider like AWS, you need to provide your cloud credentials to the container. Airnode currently only supports deploying to [AWS](https://aws.amazon.com/lambda/).

### AWS
There are two options to provide your AWS credentials to the Docker container

#### Option A: Environment variables:
Create a `aws.env` file with following content and reference it when running Docker
```
AWS_ACCESS_KEY_ID=...
AWS_SECRET_ACCESS_KEY=...
AWS_SESSION_TOKEN=...
```

```bash
$ docker run --env-file aws.env ...
```

Alternatively, you can provide the environment variables in separately to the Docker command
```bash
docker run -e AWS_ACCESS_KEY_ID=... -e AWS_SECRET_ACCESS_KEY=... ...
```

#### Option B: A configuration file:
You can use a [credential file](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where) to store toy AWS credentials. Passing this file to the container will allow the Deployer to use these credentials:
```bash
docker run -v ${HOME}/.aws/credentials:/root/.aws/credentials ...
```

## Permissions

> The Deployer is by default run by the user `root`. This may lead to some permission issues since the Deployer provides an output in a form of a `receipt.json` file. To avoid the permission problems you can specify with what [UID (user identifier)](https://en.wikipedia.org/wiki/User_identifier) and [GID (group identifier)](https://en.wikipedia.org/wiki/Group_identifier) should be the Deployer run. You can do that by setting the environment variables `USER_ID` and `GROUP_ID`:

Reviewer: ```suggestion
By default, the Deployer is run by the user `root`. This may lead to some permission issues since the Deployer provides an output in a form of a `receipt.json` file. To avoid any permission problems, you can specify the [UID (user identifier)](https://en.wikipedia.org/wiki/User_identifier) and [GID (group identifier)](https://en.wikipedia.org/wiki/Group_identifier) that the Deployer should use. You can do that by setting the environment variables `USER_ID` and `GROUP_ID`:
```

```bash
docker run -e USER_ID=$(id -u) -e GROUP_ID=$(id -g) ...
```

## Volumes
The Deployer needs two configuration files for deployment: `config.json` and `secrets.env`. After a successful deployment has been completed, a `receipt.json` will be created in a new `output/` directory. `receipt.json` must be provided for the removal of a deployed Airnode.

All of these files need to be passed to the Docker container via volumes.

> By default, the Deployer looks for configuration files in `/app/config` directory and writes the output to the `/app/output` directory.

Reviewer: I tried to make it a bit clearer that the user's local directory doesn't need to mirror the `/app` folder.
```suggestion
By default, the Docker container looks for configuration files mounted internally in the `/app/config` directory and writes output files to the `/app/output` directory. 
```

**This documentation focuses on the Deployer Docker image and its usage, not the Deployer itself. If you want learn more about the Deployer, please read [its documentation](../packages/deployer/README.md).**

# Build
You can build the Docker image by running:
```bash
docker build -f docker/Dockerfile -t api3/deployer:latest .
```

Before the build, all Deployer dependencies (but not the Deployer itself) must be available in the package registry.

> You can use build argument `yarnRegistry` to specify a different package registry for package installation. This is useful for testing a Docker build with unreleased dependencies. You can use for example [Verdaccio](https://verdaccio.org/) for this purpose.

Reviewer 2: This one seems too technical to me and would scare me a bit :D 
Maybe mark it as "advanced usage"?

Reviewer 2: You can use ">" syntax to make it look like a tip.

# Configuration
## Credentials
In order to deploy Airnode to cloud you need to pass your cloud credentials to the container. We're currently supporting only AWS cloud.

### AWS
You have two options how to pass your AWS credentials

#### Via environment variables:
You can either prepare a file with environment variables and pass that
```bash

> $ cat aws.env

Reviewer: The user might not know what `cat` does. I would say the instructions should rather say something along the lines of "create a `aws.env` file with the following contents: ..."

AWS_ACCESS_KEY_ID=...
AWS_SECRET_ACCESS_KEY=...
AWS_SESSION_TOKEN=...

$ docker run --env-file aws.env ...
```

or you can pass the environment variables separately
```bash
docker run -e AWS_ACCESS_KEY_ID=... -e AWS_SECRET_ACCESS_KEY=... ...
```

#### Via configuration file:

> You can use a [credential file](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where) to store toy AWS credentials. Passing this file to the container will allow Deployer to use these credentials:

Reviewer: ```suggestion
You can use a [credential file](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where) to store toy AWS credentials. Passing this file to the container will allow the Deployer to use these credentials:
```

```bash
docker run -v ${HOME}/.aws/credentials:/root/.aws/credentials ...
```

> ## User mapping

Reviewer: Maybe this heading should be "Permissions" or something like that instead? It's not clear what "User mapping" means

> Since Deployer provides an output in a form of a `receipt.json` file, you may find easier to run the Deployer CLI with the UID/GID you're using outside the container to avoid any permission problems. For this reason Deployer image provides environment variables `USER_ID` and `GROUP_ID` allowing you to set UID/GID for execution:

Reviewer: This sentence reads a bit weird to me. I also doubt the user will know what UID and GID mean

```bash

> docker run -e USER_ID=$(id -u) -e GROUP_ID=$(id -g) ...

Reviewer 2: Can't this be a default somehow?

Author: It can't really :confused: It has to be passed from outside (the `id` command is run on your underlying system) and it has to be passed when the container is starting.

You can run it without it but the `receipt.json` will be written to your filesystem by `root` which is usually not what you want. I mean, it doesn't break anything but it's anoying.

```

## Volumes

> Deployer needs two configuration files for deployment `config.json` and `secrets.env` and outputs `receipt.json`. It can also take `receipt.json` as an argument for deployment removal. All these files need to be passed to the container via volumes.

Reviewer: ```suggestion
The Deployer needs two configuration files for deployment: `config.json` and `secrets.env`. After a successful deployment has been completed, a `receipt.json` will be created in a new `output/` directory. `receipt.json` must be provided for the removal of a deployed Airnode.

All of these files need to be passed to the Docker container via volumes.
```

> Deployer by default looks for the configuration files in `/app/config` directory and writes the output to `/app/output` directory.

Reviewer: ```suggestion
By default, the Deployer looks for configuration files in `/app/config` directory and writes the output to the `/app/output` directory.
```

```bash
$ tree
.
├── aws.env
├── config
│   ├── config.json
│   └── secrets.env
└── output
$ docker run -v $(pwd)/config:/app/config -v $(pwd)/output:/app/output ...
$ ls output/
receipt.json
```

# Usage

> Example directory structure and commands for running Deployer Docker container. Commands are run from the depicted directory.

Reviewer: I think we should prefix every "Deployer" word with "the"

```suggestion
Example directory structure and commands for running the Deployer Docker container. The below commands are run from the depicted directory.
```

> *If you are using Windows, use CMD (and not PowerShell), replace `\` with `^` and `$(pwd)` with `%cd%`.*

Reviewer 2: Nit: Use `>`
